class-tumblr-import.php

```diff
@@ -639,16 +774,61 @@ public function do_pages_import( $url ) {
 		}
 
 		/**
-		 * Handles the sideload import.
+		 * Handles importing any Media within the imported Post
+		 *
+		 * @param  array $post the (already) imported WP_Post
+		 *
+		 * @return void
+		 */
+		public function handle_sideload( $post ) {
+			if ( empty( $post['format'] ) ) {
+				return; // Nothing to import
+			}
+
+			$sideload_method = 'handle_sideload_' . $post['format'] . '_post';
+
+			if ( ! method_exists( $this, $sideload_method ) ) {
+				return;
+			}
+
+			return $this->{$sideload_method}( $post );
+		}
+
+		/**
+		 * Handles updating the sideloaded post
+		 * Called if the handler function successfully completes
+		 * and is able to retrieve a valid $post
 		 *
-		 * @param array  $post The post.
-		 * @param string $source The source.
+		 * @param  array $post The WP_Post to update
+		 *
+		 * @return void
+		 */
+		public function handle_sideload_post_update( $post ) {
+			do_action( 'tumblr_importer_format_post_before', $post );
+			$post = apply_filters( 'tumblr_importer_format_post', $post );
+			do_action( 'tumblr_importer_format_post_after', $post );
+
+			do_action( 'tumblr_importer_metadata_before', $post );
+			do_action( 'tumblr_importer_metadata', $post );
+			do_action( 'tumblr_importer_metadata_after', $post );
+
+			do_action( 'tumblr_importer_sideload_wp_update_post_before', $post );
+			wp_update_post( $post );
```

Comment:
Same here, I think turning off hooks would be a good idea

https://developer.wordpress.org/reference/functions/wp_update_post/